lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -386,6 +386,25 @@ walletStateOptions = taggedSumTypeOptions $ TaggedObjectOptions
     , _contentsFieldName = "progress"
     }
 
+{-------------------------------------------------------------------------------
+                             FromText/ToText instances
+-------------------------------------------------------------------------------}
+
+instance FromText AddressAmount where
+    fromText text = do
+        let err = Left . TextDecodingError $ "Parse error. Expecting format\
+            \'<amount>@<address>' but got '" <> show text <> "'"
+        case split (=='@') text of
+            [] -> err
+            [_] -> err
+            -- TODO: more user friendly message
```

Comment:
Avoid TODOs, put an idea on the blackboard such that we think about discussing it :)